### PR TITLE
Add piped input tests for RevertSam, MergeBamAlignment

### PR DESCRIPTION
**Pull request by takutosato**
### Description

When we cloudify a Picard tool, we change the input and output file types from `FILE` to `PicardHtsPath.` This could trigger a "java.io.IOException: Illegal seek" if the input SAM is piped (/dev/stdin). I believe this is related to a known issue (see samtools/htsjdk#1084, samtools/htsjdk#1077). The workaround is to wrap a Path object in `SamInputResource.of()`, which was put in place in samtools/htsjdk#1124. The tests in PR should safeguard us from inadvertently removing the workaround until a more robust fix can be implemented.

The targeted tools are RevertSam and MergeBamAlignment, both of which are used with piped input in warp pipelines. For example, see
https://github.com/broadinstitute/warp/blob/develop/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl
https://github.com/broadinstitute/warp/blob/develop/tasks/broad/Alignment.wdl

To reproduce the bug, change Line 272 of RevertSam from:

```java 
final SamReader in = SamReaderFactory.makeDefault().referenceSequence(referenceSequence.getReferencePath()).validationStringency(VALIDATION_STRINGENCY).open(SamInputResource.of(INPUT.toPath())); ;
```

to 

```java
final SamReader in = SamReaderFactory.makeDefault().referenceSequence(referenceSequence.getReferencePath()).validationStringency(VALIDATION_STRINGENCY).open(INPUT.toPath());
```

That is, remove the wrapper `SamInputResource.of()`.

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

#### Content
- [x] Added or modified tests to cover changes and any new functionality
- [x] Edited the README / documentation (if applicable)
- [x] All tests passing on github actions

#### Review
- [ ] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

### CODE REVIEW

File: src/main/java/picard/sam/RevertSam.java [hunk 1]
```diff
@@ -269,7 +269,7 @@ protected int doWork() {
         }
 
         final boolean sanitizing = SANITIZE;
-        final SamReader in = SamReaderFactory.makeDefault().referenceSequence(referenceSequence.getReferencePath()).validationStringency(VALIDATION_STRINGENCY).open(SamInputResource.of(INPUT.toPath())); // tsato: confirm this won't break piped input
+        final SamReader in = SamReaderFactory.makeDefault().referenceSequence(referenceSequence.getReferencePath()).validationStringency(VALIDATION_STRINGENCY).open(SamInputResource.of(INPUT.toPath()));
```
Comment (lbergelson) [inline src/main/java/picard/sam/RevertSam.java:272]: @takutosato Could you put a comment here referencing this PR or the related issue?  Just too save us some time when we forget and someone tries to change it.

**Review by lbergelson — APPROVED** [review]
I assume you tested that the tests fail when you don't have that work around in place One request for additional comment and then good to merge.  Thank you!

File: src/main/java/picard/sam/RevertSam.java (hunk 1 above)
Comment (takutosato) [reply src/main/java/picard/sam/RevertSam.java:278]: Done

**Comment by takutosato** [comment]
@lbergelson yes the test fails when the workaround is removed.

File: src/main/java/picard/sam/RevertSam.java (hunk 1 above)
Comment (lbergelson) [reply src/main/java/picard/sam/RevertSam.java:278]: 👍

**Review by lbergelson — APPROVED** [review]
👍

**Pull request merged**